### DISCUSSION TITLE
Updates from LiveCD/USB usually fail

### DISCUSSION THREAD

**Discussion by apodtele** [fwupd/fwupd#7367] [General]
It probably does not make sense at first. Why would you want to update firmware using LiveUSB if you don't intend to install it? Anyhow, this is what you might get using Fedora 40 Live CD
```
/run/media/root/ANACONDA does not have sufficient space, required 50,3 MB, got 475,1 kB
```

USB or RAM usually has plenty of space to use....

It would be cool if fwupd was available on GParted Live distribution. Or, perhaps, distribute its own fwupd LiveUSB.

**Comment by superm1** [comment]
You're meaning specifically UEFI firmware updates, right?  You need an EFI system partition to trigger the update.  If the system already has one I would expect fwupd would have used it instead of falling back to the USB key you were booted from.

**Reply by apodtele** [reply]
It failed to update an Ethernet controller in my particular case, it failed to uncompress it. Are you saying that an EFI partition is the only reliable space available and it is just too tight on my system?

**Reply by superm1** [reply]
Ethernet controller shouldn't need space like that.  To characterize what's going on, I suggest you restart the daemon in verbose mode on the live media and then share the daemon log while it reproduces.

**Comment by nipsy** [comment]
Just to add my two cents here, I was wanting to be able to run fwupdmgr in such an environment on a bare system with a zeroed out drive.  Getting to the point where this actually worked took some effort, so I figured I'd document it here in case anyone else stumbles across this discussion.  This could also be tangentially related to #8853, but I won't swear to that as I'm using an entirely different distribution.

First things first, I'm creating an ISO in NixOS with the necessary ```services.fwupd.enable = true;``` and ```services.udisks2.enable = true;``` set.  This doesn't actually function right out of the box at boot, but it creates the systemd service entries which will be helpful shortly.

So once I was booted, the first thing I needed to do was create the EFI System partition in fdisk.  I used GPT for the disk label and created a 1 GB ESP at the start of the drive.  After writing that and quitting in fdisk, I formatted it with ```mkfs.vfat /dev/nvme0n1p1``` and then went to mount it.  Most importantly, the only place (at least for this version of fwupd in NixOS) that worked correctly was ```mount /dev/nvme0n1p1 /boot``` which of course required a ```mkdir /boot``` beforehand and after it was mounted, a subsequent ```mkdir /boot/EFI```.  I also tried ```/boot/efi``` and ```/esp``` as the initial mount point, but fwupd couldn't find either of those in this scenario.  ```/boot/EFI``` is entirely empty at this point, but fwupd does end up using this directory correctly as expected once the commands below are run.

Next thing was to ```systemctl restart udisks2.service fwupd.service``` as the former isn't actually started at boot in this environment and the second presumably needs the first running to work at all.

At last, a ```fwupdmgr refresh``` (with an established network connection obviously; not including that part here) was happy and seeing everything and a ```fwupdmgr update``` did its magic.

Now, one last problem I ran into was on the reboot to flash everything.  If I left the USB key in the system, the flash failed and I saw some ```file system full``` error.  I can't say that's the exact wording, but it was an equivalent error.  I had to remove the USB key as the system rebooted after the update command to have it flash correctly.  I assume the USB drive being present was confusing the fwupdx64.efi loader.

I just upgraded my ThinkPad L14 Gen 1 with the latest BIOS and secure boot keys present upstream with this approach.  The ESP on the internal NVMe drive still contains the aforementioned EFI application under ```/boot/EFI/nixos``` with a cap file present under ```/boot/EFI/nixos/fw``` also.  Good enough for me.  Easy to clean all that up after the fact if I want and I don't need to do even a minimal install on the actual NVMe drive to flash a supported system.

Hopefully this helps others.  At the very least, I'll know for myself how to do it in future.
